src/marlin_stubs/pause/pause.cpp

```diff
@@ -739,20 +740,20 @@ void Pause::park_nozzle_and_notify() {
     //home the X or Y axis if it is not homed and we want to move it
     //homing is after Z move to be clear of all obstacles
     //Should not affect other operations than Load/Unload/Change filament run from home screen without homing. We are homed during print
-    // FIXME: `current_position.x - park_pos.x != 0` is not a safe condition because `current_position` is unreliable (not homed) use NAN instead.
-    // TODO: make homeaxis non-blocking to allow quick_stop
-    if ((axis_homed & _BV(X_AXIS)) == 0 && current_position.x - park_pos.x != 0) {
-        homeaxis(X_AXIS);
-    }
-    if ((axis_homed & _BV(Y_AXIS)) == 0 && current_position.y - park_pos.y != 0) {
-        homeaxis(Y_AXIS);
+    LOOP_XY(axis) {
+        // TODO: make homeaxis non-blocking to allow quick_stop
+        if (!isnan(park_pos.pos[axis]) && (axis_homed & axis_known_position & _BV(axis)) == 0) {
+            GcodeSuite::G28_no_parser(false, false, 0, false, axis == X_AXIS, axis == Y_AXIS, false);
```

Comment:
I think we can not use G28 here, as it resets MBL. And we definitely can not reset MBL while parking nozzle.